percona/pmm/pmm.go

```diff
@@ -412,17 +419,30 @@ func sidecarContainerV3(pgc *v2.PerconaPGCluster) corev1.Container {
 			},
 			{
 				Name:  "PMM_AGENT_PRERUN_SCRIPT",
-				Value: agentPrerunScript(pgc.Spec.PMM.QuerySource),
+				Value: agentPrerunScript(pgc.Spec.PMM.QuerySource, pgc),
 			},
 			{
 				Name:  "PMM_AGENT_PATHS_TEMPDIR",
 				Value: "/tmp",
 			},
 		},
 	}
+
+	if pgc.CompareVersion("2.7.0") >= 0 {
+		clusterName := pgc.Name
+		if pgc.Spec.PMM.CustomClusterName != "" {
+			clusterName = pgc.Spec.PMM.CustomClusterName
+		}
+		container.Env = append(container.Env, corev1.EnvVar{
+			Name:  "CLUSTER_NAME",
+			Value: clusterName,
+		})
+	}
```

Comment:
Do we need to do this check for PMM3? The related implementation was merged very recently so I think we can skip checking the cr version for it.

Reply:
not sure yet. @egegunes what do you think?

Reply:
I deleted the condition but keep container variable. Next time when we need to add variable under version we can easily do it. But if it's not ok, I can delete it and just return.

Reply:
it is fine for me, minor styling.